packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3073,7 +3073,16 @@ class _RouteEntry extends RouteTransitionRecord {
           entry.removeListener(listener);
           if (mounted == 0) {
             assert(route.overlayEntries.every((OverlayEntry e) => !e.mounted));
-            route.dispose();
+            // This is a listener callback of one of the overlayEntries in this
+            // route. Disposing the route also disposes its overlayEntries and
+            // violates the rule that a change notifier can't be disposed during
+            // its notifying callback.
+            //
+            // Use a microtask to ensure the overlayEntries have finished
+            // notifying their listeners before disposing.
+            scheduleMicrotask(() {
```

Comment:
I am actually run into a dead end here.

This microtask causes issue where navigator.dispose expects all route are disposed. I think we may need a way to clean up the entry out of term while destroying all the resource other than the entry's changenotifier.